node_monitor/node_monitor.py

```diff
@@ -173,6 +175,35 @@ def broadcast_status_report(self) -> None:
                     if err2 is not None:
                         logging.error(f"TelegramBot.send_message() failed with error: {err2}")
             # - - - - - - - - - - - - - - - - -
+    
+
+    def update_node_provider_lookup_if_new(
+            self, 
+            override_api_data: ic_api.NodeProviders | None = None) -> None:
+        """Fetches the current node providers from the ic-api and compares
+        them to what is currently in the node_provider_lookup table in the 
+        database. If there is a new node provider in the API, they will be
+        added to our database.
+
+        Args:
+            override_data: If provided, this arg will be used instead of 
+                live fetching Node Providers from the ic-api. Useful for testing.
+        """
```

Comment:
1. Why an override? This function mutates data in the db too btw.

Reply:
I chose an override as it is similar to what you did for the `_resync()` function. This way we can test the function without making a call to the API each timer. 

The tests (`test_no_new_node_provider` and `test_one_new_node_provider` in `test_node_monitor.py`) do not actually insert data into the database. It uses a mock instance of the database just to check if the function gets called. Should I remove these automated tests then?

Reply:
Good call, I forgot that we've got a mock function here. Thanks.